trie/trie_reader.go

```diff
@@ -27,19 +27,19 @@ import (
 // for concurrent usage.
 type trieReader struct {
 	owner  common.Hash
-	reader database.Reader
+	reader database.NodeReader
 	banned map[string]struct{} // Marker to prevent node from being accessed, for tests
 }
 
 // newTrieReader initializes the trie reader with the given node reader.
-func newTrieReader(stateRoot, owner common.Hash, db database.Database) (*trieReader, error) {
+func newTrieReader(stateRoot, owner common.Hash, db database.NodeDatabase) (*trieReader, error) {
 	if stateRoot == (common.Hash{}) || stateRoot == types.EmptyRootHash {
 		if stateRoot == (common.Hash{}) {
 			log.Error("Zero state root hash!")
 		}
 		return &trieReader{owner: owner}, nil
 	}
-	reader, err := db.Reader(stateRoot)
+	reader, err := db.NodeReader(stateRoot)
 	if err != nil {
 		return nil, &MissingNodeError{Owner: owner, NodeHash: stateRoot, err: err}
```

Comment:
Could you add the same comment that the return value should not be modified to trieReader.node ? 
Afaict theres a lot of usages of this that seem a bit unsafe (esp. in Verkle). Maybe it makes sense to do a deep copy here?

Reply:
done

Reply:
> Maybe it makes sense to do a deep copy here?

I think it will introduce non-trivial overhead, i would prefer to do the copy by the caller itself (e.g. verkle) for some special use cases